docs/DeveloperGuide.md

```diff
@@ -1,38 +1,527 @@
 # Developer Guide
 
+## Table of Contents
+
+* [Acknowledgements](#acknowledgements)
+* [Introduction](#introduction)
+* [Design](#design)
+* [Implementation](#implementation)
+* [Requirements](#requirements)
+    * [Project Scope](#project-scope)
+    * [User Stories](#user-stories)
+    * [Non-functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+## Introduction
+
+The aim of this guide is to provide a comprehensive explanation of the internal mechanisms of MoneyGoWhere.
+This will enable upcoming software engineers to acquire a detailed understanding of the application's implementation,
+making it easier for them to work on the project.
+
+For all sequence diagrams included in this guide, we will not include the interaction 
+with the Parser and Router classes as it is implied that all inputs will first be parsed through them. 
+
+## Design
+
+## Implementation
+
+### Item Commands
+
+#### Overview
+
+* [Add an Item](#add-an-item)
+  * [Advanced Mode](#advanced-mode-add-an-item)
+  * [Basic Mode](#basic-mode-add-an-item)
+* [Delete an Item](#delete-an-item)
+  * [Advanced Mode](#advanced-mode-delete-an-item)
+  * [Basic Mode](#basic-mode-delete-an-item)
+* [List all Items](#list-all-items)
+* [Update an Item](#update-an-item)
+    * [Advanced Mode](#advanced-mode-update-an-item)
+    * [Basic Mode](#basic-mode-update-an-item)
+* [Find an Item](#find-an-item)
+  * [Advanced Mode](#advanced-mode-find-an-item)
+  * [Basic Mode](#basic-mode-find-an-item)
+
+<hr>
+
+#### Add an Item
+
+##### Advanced Mode Add an Item
+The expected inputs to add only one menu item into an order is as such:
+* `/additem -n <item_name> -p <price>`
+
+This sequence diagram models the interaction between various components in MoneyGoWhere when the user inputs the 
+command `/additem`.
+
+![](./images/developersGuide/addItem.png)
```

Comment:
ui.promptUserInput() recives the input of the User. Maybe the "additem order" sent by User should be sent to the UI？